src/lazy/encoder/text/v1_1/value_writer.rs

```diff
@@ -271,31 +291,47 @@ pub struct TextEExpWriter_1_1<'value, W: Write> {
     // There is no e-exp writer in 1.0 to which we can delegate,
     // but we can re-use the TextContainerWriter_1_0 for a lot of the formatting.
     container_writer: TextContainerWriter_1_0<'value, W>,
-    // TODO: Hold a reference to the macro signature and advance to the next parameter on
-    //       each method call. Any time a value is written, compare its type to the parameter to
-    //       make sure it's legal.
+    macros: &'value MacroTable,
+    signature_iter: SignatureIterator<'value>,
+}
+
+#[test]
+fn say() {
+    println!("{}", size_of::<TextContainerWriter_1_0<'_, Vec<u8>>>());
+    println!("{}", size_of::<MacroIdRef<'_>>());
 }
 
 impl<'value, W: Write> TextEExpWriter_1_1<'value, W> {
     pub(crate) fn new(
         writer: &'value mut LazyRawTextWriter_1_0<W>,
         depth: usize,
         parent_type: ParentType,
-        container_type: ContainerType,
-        opening_delimiter: &str,
-        value_delimiter: &'static str,
-        trailing_delimiter: &'static str,
+        opening_text: &str,
+        macros: &'value MacroTable,
+        invoked_macro: MacroRef<'value>,
```

Comment:
🪧 Adding parameters to this function (to plumb through the macro information) caused it to trip `clippy`'s alarm for excessive parameters. This is only called in one place, so I refactored it to eliminate some parameters.